examples/js/loaders/GLTF2Loader.js

```diff
@@ -2531,28 +2531,34 @@ THREE.GLTF2Loader = ( function () {
 		var extensions = this.extensions;
 		var scope = this;
 
-		return _each( json.nodes, function ( node ) {
-
-			var matrix = new THREE.Matrix4();
+		var nodes = json.nodes || [];
 
-			var _node;
+		// Nothing in the node definition indicates whether it is a Bone or an
+		// Object3D. So, traverse the hierarchy once in advance, using node.skins
+		// references to mark bones.
+		nodes.forEach( function ( node ) {
```

Comment:
@donmccurdy 
Why do you traverse all nodes and check each for `skin`? It would be faster to traverse skins directly:
```javascript
skins.forEach( function ( skin ) {

    if ( skin.joints !== undefined ) {

        skin.joints.forEach( function ( id ) {

            nodes[ id ].isBone = true;

        } );

    }
    
} );
```